src/PlaneWaveBasis.jl

```diff
@@ -332,32 +319,77 @@ end
 
 
 """
-Return the list of wave vectors (integer coordinates) for the cubic basis set.
+    G_vectors(fft_size::Tuple)
+
+The wave vectors `G` in reduced (integer) coordinates for a cubic basis set
+of given sizes.
 """
-function G_vectors(fft_size)
+function G_vectors(fft_size::Tuple)
     start = .- cld.(fft_size .- 1, 2)
     stop  = fld.(fft_size .- 1, 2)
     axes = [[collect(0:stop[i]); collect(start[i]:-1)] for i in 1:3]
     (Vec3{Int}(i, j, k) for i in axes[1], j in axes[2], k in axes[3])
 end
+
+@doc raw"""
+    G_vectors(basis::PlaneWaveBasis)
+    G_vectors(kpt::Kpoint)
+
+The list of wave vectors ``G`` in reduced (integer) coordinates of a `basis`
+or a ``k``-point `kpt`.
+"""
 G_vectors(basis::PlaneWaveBasis) = G_vectors(basis.fft_size)
+G_vectors(kpt::Kpoint) = kpt.G_vectors
 
+
+@doc raw"""
+    G_vectors_cart(basis::PlaneWaveBasis)
+    G_vectors_cart(basis::PlaneWaveBasis, kpt::Kpoint)
+
+The list of ``G`` vectors of a given `basis` or `kpt`, in cartesian coordinates.
+"""
 function G_vectors_cart(basis::PlaneWaveBasis)
     (basis.model.recip_lattice * G for G in G_vectors(basis.fft_size))
 end
+function G_vectors_cart(basis::PlaneWaveBasis, kpt::Kpoint)
+    (basis.model.recip_lattice * G for G in G_vectors(kpt))
+end
```

Comment:
if we do that shouldn't we just have basis, kpt for G_vectors for consistency?

Reply:
I wondered about that myself. Given that we both wonder I do that for consistency.